### PR TITLE
overlord/devicestate: preserve the current model inside ubuntu-boot

### PR DESCRIPTION
**Pull request by bboozzoo**
Preserve the installed model inside ubuntu-boot partition. We will need to
measure the model attributes in initramfs, this change enables that.

### CODE REVIEW

File: overlord/devicestate/devicestate_install_mode_test.go [hunk 1]
```diff
@@ -215,6 +216,9 @@ func (s *deviceMgrInstallModeSuite) doRunChangeTestWithEncryption(c *C, grade st
 	c.Assert(modeenv.WriteTo(""), IsNil)
 	devicestate.SetSystemMode(s.mgr, "install")
 
+	err := os.MkdirAll(boot.InitramfsUbuntuBootDir, 0755)
```
Comment (pedronis) [inline overlord/devicestate/devicestate_install_mode_test.go:219]: for realism if I understand correctly, this should be done from inside the mocked snap-boostrap

File: overlord/devicestate/devicestate_install_mode_test.go [hunk 2]
```diff
@@ -392,7 +396,12 @@ func (s *deviceMgrInstallModeSuite) mockInstallModeChange(c *C, modelGrade strin
...
```
Comment (pedronis) [inline overlord/devicestate/devicestate_install_mode_test.go:399]: same

File: overlord/devicestate/devicestate_install_mode_test.go (hunk 1 above)
Comment (bboozzoo) [reply overlord/devicestate/devicestate_install_mode_test.go:219]: Yes. I'll leave a comment here.

File: overlord/devicestate/devicestate_install_mode_test.go [hunk 3]
```diff
@@ -471,3 +482,24 @@ func (s *deviceMgrInstallModeSuite) TestInstallModeNoCloudInitForSigned(c *C) {
 		{TargetRootDir: boot.InitramfsWritableDir},
 	})
 }
+
+func (s *deviceMgrInstallModeSuite) TestInstallModeWritesModel(c *C) {
+	// pretend we have a cloud-init config on the seed partition
```
Comment (mvo5) [inline overlord/devicestate/devicestate_install_mode_test.go:487]: Copy/pate issue? This test is not about cloud-init ?

File: overlord/devicestate/handlers_install.go [hunk 4]
```diff
@@ -109,6 +118,12 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 		return fmt.Errorf("cannot create partitions: %v", osutil.OutputErr(output, err))
 	}
 
+	// keep track of the model we installed
+	err = writeModel(deviceCtx.Model(), filepath.Join(boot.InitramfsUbuntuBootDir, "model"))
```
Comment (mvo5) [inline overlord/devicestate/handlers_install.go:122]: (nitpick^2) this could be line with "if err := ..."

**Pull request merged**